dice_ml/explainer_interfaces/explainer_base.py

```diff
@@ -392,14 +390,13 @@ def do_posthoc_sparsity_enhancement(self, final_cfs_sparse, query_instance, post
 
         cfs_preds_sparse = []
 
-        for cf_ix in range(len(final_cfs_sparse)):
-            current_pred = self.predict_fn_for_sparsity(
-                final_cfs_sparse.iloc[[cf_ix]][self.data_interface.feature_names])
+        for cf_ix in list(final_cfs_sparse.index):
+            current_pred = self.predict_fn_for_sparsity(final_cfs_sparse.loc[[cf_ix]][self.data_interface.feature_names])
             for feature in features_sorted:
                 # current_pred = self.predict_fn_for_sparsity(final_cfs_sparse.iat[[cf_ix]][self.data_interface.feature_names])
                 # feat_ix = self.data_interface.continuous_feature_names.index(feature)
-                diff = query_instance[feature].iat[0] - final_cfs_sparse[feature].iat[cf_ix]
-                if abs(diff) <= quantiles[feature]:
+                diff = query_instance[feature].iat[0] - int(final_cfs_sparse.loc[[cf_ix]][feature])
```

Comment:
same comment. `at` is the analogue of `iat`, just like loc for iloc. `at` is much faster than `loc` when retrieving a single row/value, so let's use that everywhere.